src/hotspot/share/opto/macro.hpp

```diff
@@ -196,7 +193,7 @@ class PhaseMacroExpand : public Phase {
                           Node* klass_node, Node* length,
                           Node* size_in_bytes);
 
-  Node* make_arraycopy_load(ArrayCopyNode* ac, intptr_t offset, Node* ctl, Node* mem, BasicType ft, const Type *ftype, AllocateNode *alloc);
+  static Node* make_arraycopy_load(Compile* comp, PhaseIterGVN* igvn, ArrayCopyNode* ac, intptr_t offset, Node* ctl, Node* mem, BasicType ft, const Type *ftype, AllocateNode *alloc);
```

Comment:
Why you need this change? It polluted diffs and hide important changes. Could be separate change from this one.

Reply:
I had to make this method static because it uses `value_from_mem` - which I also made static. I had to make `value_from_mem` static so that I can use it outside PhaseMacroExpand.

Reply:
I see, you use in escape.cpp. Okay. I need to review changes there too.

Reply:
or you could construct a temporary PhaseMacroExpand object in EA. 

I see that you convert many member function to static so you can query in EA. the only blocker is _igvn.

Reply:
That seems a good idea. Together with some other refactoring I decided to revert making the methods static and instead use them through an instance of ME.